docs/050-breaking-changes.rst

```diff
@@ -0,0 +1,85 @@
+********************************
+Solidity v0.5.0 Breaking Changes
+********************************
+
+This section highlights the main breaking changes introduced in Solidity
+version 0.5.0, along with the reasoning behind the changes and how to update
+affected code.
+For the full list check
+`the release changelog <https://github.com/ethereum/solidity/releases>`_.
+
+Semantic Only Changes
+=====================
+
+This section lists the changes that are semantic-only, thus potentially
+hiding new and different behaviors in previous code.
+
+* Signed right shift now uses proper arithmetic shift, i.e. rounding towards
+  negative infinity. Signed and unsigned shift will have dedicated opcodes in
+  Constantinople, and are emulated by Solidity for the moment.
+
+* The ``continue`` statement in a ``do...while`` loop now jumps to the
+  condition, which is the common behavior in such cases. It used to jump to the
+  loop body.
+
+Semantic and Syntactic Changes
+==============================
+
+This section highlights changes that affect syntax and semantics.
+
+* The functions ``.call()`` (and family), ``keccak256()``, ``sha256()``,
+  ``ripemd160()`` now accept only a single ``bytes`` argument. Moreover, the
+  argument is not padded.  TODO: write reason here.  Change every ``.call()``
+  to a ``.call("")`` and every ``.call(signature, a, b, c)`` to use
+  ``.call(abi.encodeWithSignature(signature, a, b, c))`` (the last one only
+  works for value types).  Change every ``keccak256(a, b, c)`` to
+  ``keccak256(abi.encodePacked(a, b, c))``.
```

Comment:
Something that is not really a breaking change but perhaps might be useful for people to know:
Change `x.call(bytes4(keccak256("f(uint256)"), a, b)` to `x.call(abi.encodeWithSignature("f(uint256)", a, b))`. - i.e. explicitly show that they do not need to do the keccak-dance anymore.